File: zig/src/types/Renderer.zig
const std = @import("std");

pub const Renderer = struct {
    const Self = @This();

    const MarkdownNode = @import("MarkdownNode.zig").MarkdownNode;
    const RendererState = @import("RendererState.zig").RendererState;

    name: []const u8,

    render: *const fn (
        node: *const MarkdownNode,
        state: *RendererState,
        first: ?bool,
        last: ?bool,
        decode: ?bool,
    ) callconv(.Inline) void,
};

File: swift/Sources/Allmark/Types/Renderer.swift
import Foundation

/// A renderer that converts an AST node to output.
@MainActor
public struct Renderer {
	/// The renderer name (matches the node type it renders).
	public var name: String
	/// The render function.
	public var render: @MainActor (MarkdownNode, inout RendererState, Bool?, Bool?, Bool?) -> Void

	public init(name: String, render: @MainActor @escaping (MarkdownNode, inout RendererState, Bool?, Bool?, Bool?) -> Void) {
		self.name = name
		self.render = render
	}
}
